src/components/MapView/index.tsx

```diff
@@ -80,33 +82,75 @@ function MapView({ classes }: MapViewProps) {
   }, [dispatch]);
 
   useEffect(() => {
     // when we switch layers, close any active pop-ups
     dispatch(hidePopup());
-  }, [dispatch, layers]);
-
-  const serverLayers = layers.filter((layer): layer is
-    | WMSLayerProps
-    | ImpactLayerProps => ['impact', 'wms'].includes(layer.type));
-
-  const selectedLayerDates = uniq(
-    serverLayers
-      .map(layer =>
-        layer.type === 'wms'
-          ? serverAvailableDates[layer.serverLayerName]
-          : serverAvailableDates[
-              (LayerDefinitions[layer.hazardLayer] as WMSLayerProps)
-                .serverLayerName
-            ],
-      )
-      .filter(value => value)
-      .flat(),
-  );
+  }, [dispatch, selectedLayers]);
+
+  // calculate possible dates user can pick from the currently selected layers.
+  // with useMemo we save on performance and only show the notification when layers selected amount changes, making it a useEffect too.
+  const selectedLayerDates: number[] = useMemo(() => {
+    const dateSupportLayerTypes: Array<LayerType['type']> = [
+      'impact',
+      'point_data',
+      'wms',
+    ];
+    const layersWithDateSupport = selectedLayers.filter((layer): layer is
+      | WMSLayerProps
+      | ImpactLayerProps
+      | PointDataLayerProps => dateSupportLayerTypes.includes(layer.type));
+    if (layersWithDateSupport.length === 0) return [];
+
+    /*
+       takes all the dates possible for every layer and counts the amount of times each one is duplicated.
+       if a date's duplicate amount is the same as the number of layers active, then this date is compatible with all layers selected.
+    */
+    const selectedLayerDatesDupCount = countBy(
+      layersWithDateSupport
+        // the 2 below eslint disables are justified as TS will strongly ensure undefined is never returned from map()
+        // eslint-disable-next-line array-callback-return,consistent-return
+        .map((layer): number[] => {
+          // eslint-disable-next-line default-case
+          switch (layer.type) {
+            case 'wms':
+              return serverAvailableDates[layer.serverLayerName];
+            case 'impact':
+              return serverAvailableDates[
+                (LayerDefinitions[layer.hazardLayer] as WMSLayerProps)
+                  .serverLayerName
+              ];
+            case 'point_data':
+              return serverAvailableDates[layer.id];
+          }
+        })
+        .filter(value => value) // null check
+        .flat(),
+    );
+    /*
+      Only keep the dates which were duplicated the same amount of times as the amount of layers active...and convert back to array.
+     */
+    const ret = Object.keys(
+      pickBy(
+        selectedLayerDatesDupCount,
+        dupTimes => dupTimes >= layersWithDateSupport.length,
+      ),
+    ).map(Number); // convert back to number array - countBy converted all the numbers to strings
+    if (ret.length === 0) {
+      dispatch(
+        addNotification({
```

Comment:
Couldn't find a way to test this as all existing date layers have overlap (wms, impact & point_data)